plugins/link/link.cpp

```diff
@@ -228,9 +231,7 @@ void mumble_shutdownPositionalData() {
 	pluginContext.clear();
 	pluginIdentity.clear();
 
-	lm->uiTick = last_tick = 0;
-	lm->uiVersion          = 0;
-	lm->name[0]            = 0;
+	sharedMem.fillWithZero();
```

Comment:
I think this might (theoretically) be dangerous to do as we don't necessarily know that a bit representation of all zeros is valid for the types used as members in `LinkedMem`.
Maybe it would be safer to define all members inside the `LinkedMem` struct with a default value (of zero) and then add a function `reset()` to `SharedMemory` that simply looks like
```cpp
void SharedMemory::reset() const {
	LinkedMem defaultObj{};
	write(defaultObj);
}
```

Reply:
That's a great idea. I didn't add explicit default values to the struct definition because it seems like zero initialization is safe to rely on here.

Reply:
It probably is, but in order to make this kind of stuff future proof, my experience is that explicit is usually safer than implicit :)

Reply:
Okay, I added default member initializers. I made some very subjective decisions making this change, so I'll be interested to hear what you think.